### PR TITLE
First iteration of simulate validation (user operation)

### PR DESCRIPTION
**Pull request by Vid201**
This adds first checks for simulate validation - forbidden opcodes. It also removes `sanity_check_results` and rertieves the stake info at the start (alongside JS trace).

### CODE REVIEW

File: src/uopool/services/sanity_check.rs [hunk 1]
```diff
@@ -296,17 +240,9 @@ where
...
-        self.sanity_check_results.write().insert(
-            user_operation.hash(entry_point, &self.chain_id),
-            Default::default(),
-        );
-
         // Either the sender is an existing contract, or the initCode is not empty (but not both)
         self.sender_or_init_code(user_operation).await?;
 
-        // If initCode is not empty, parse its first 20 bytes as a factory address. Record whether the factory is staked, in case the later simulation indicates that it needs to be. If the factory accesses global state, it must be staked - see reputation, throttling and banning section for details.
-        self.verify_factory(user_operation, entry_point).await?;
```
Comment (zsluedem) [inline src/uopool/services/sanity_check.rs:308]: Curious why this is removed.

**Review by zsluedem — COMMENTED** [review]
I am curious what stuffs left to be resolved

**Comment by Vid201** [comment]
> I am curious what stuffs left to be resolved

You mean in general or in the simulation? For simulation, there are still several things: storage access, limitations on CALL opcodes ... I will continue with these things.

File: src/uopool/services/sanity_check.rs (hunk 1 above)
Comment (Vid201) [reply src/uopool/services/sanity_check.rs:308]: The only thing that was verified here is if the factory is staked. This is now available here: https://github.com/Vid201/aa-bundler/blob/feat/uo_trace/src/uopool/services/simulation.rs#L98, so no need to check that here also.

**Review by zsluedem — APPROVED** [review]
LGTM

**Pull request merged**